playbooks/sync-config-files.yaml

```diff
@@ -0,0 +1,96 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+---
+- name: Install hub both in nodepool and zuul
+  hosts: nodepool:zuul
+  become: yes
+  tasks:
+    - name: Install hub
+      shell:
+        cmd: |
+          ## check hub installed or not
+          if hub version >> /dev/null;then
+              echo "hub has been installed!"
+          else
+              echo "hub installing..."
+              wget https://github.com/github/hub/releases/download/v2.10.0/hub-linux-amd64-2.10.0.tgz
+              sudo tar -C /usr/local/src -xzf hub-linux-amd64-2.10.0.tgz
+              cd /usr/local/src/hub-linux-amd64-2.10.0
+              sudo ./install
+              hub version
+              echo "hub installed success!"
+          fi
+        executable: /bin/bash
+
+- name: Install other tools in nodepool
+  hosts: nodepool
+  become: yes
+  tasks:
+    - name: Install shyaml,ansible and ruamel.yaml
+      pip:
+        name: ['shyaml', 'ruamel.yaml', 'ansible']
+
+- name: Copy script to zuul nodes
+  become: yes
+  hosts: zuul
+  tasks:
+    - name: Copy zuul_files_sync.sh into place.
+      copy:
+        dest: /home/ubuntu/zuul_files_sync.sh
+        src: "{{ labkeeper_src_dir }}/labsync/zuul_files_sync.sh"
+
+
+- name: Zuul files sync script
+  hosts: zuul
+  become: yes
+  vars_files:
+    - "{{ labkeeper_src_dir }}/labsync/labsync_vars.yaml"
+  tasks:
+    - name: Call config file sync script
+      shell:
+        cmd: |
+          # add cron to create pr to github at utc 17:00 every day
+          touch sync_cron.cron
+          echo "* 17 * * * bash /home/ubuntu/zuul_files_sync.sh {{ github_username }} {{ github_useremail }} {{ github_password }} {{ deploy_type }} >> /var/log/sync.log 2>&1" >>  sync_cron.cron
+          crontab sync_cron.cron
+
+        executable: /bin/bash
+
+- name: Copy script to nodepool
+  become: yes
```

Comment:
you'd better use 1 playbook if there isn't any specific requirement. you can merge L34-L40 and L68-L96 to one play including multiple tasks. the same as the zuul plays

Reply:
ok, will refactor it, but the pip installs for nodepool should be a play(tasks in a play is unordered? right?), to make sure the scripts following to execute correctly,